bridges/ethMultiversX/bridgeExecutor.go

```diff
@@ -531,50 +524,51 @@ func (executor *bridgeExecutor) PerformTransferOnEthereum(ctx context.Context) e
-		isMintBurnToken, err := executor.isMintBurnToken(ctx, erc20Address)
+func (executor *bridgeExecutor) checkCumulatedTransfers(ctx context.Context, tokens []common.Address, convertedTokens [][]byte, amounts []*big.Int) error {
+	for i, token := range tokens {
+		err := executor.checkToken(ctx, token, convertedTokens[i], amounts[i])
 		if err != nil {
 			return err
 		}
+	}
+	return nil
+}
 
-		if isMintBurnToken {
-			err = executor.checkRequiredMintBurnBalance(ctx, erc20Address)
-		} else {
-			err = executor.ethereumClient.CheckRequiredBalance(ctx, erc20Address, value)
-		}
+func (executor *bridgeExecutor) checkToken(ctx context.Context, token common.Address, convertedToken []byte, amount *big.Int) error {
+	isMintBurnToken, err := executor.isMintBurnToken(ctx, token, convertedToken)
+	if err != nil {
+		return err
+	}
 
-		if err != nil {
-			return err
-		}
+	if isMintBurnToken {
+		return executor.checkRequiredMintBurnBalance(ctx, token, convertedToken)
 	}
-	return nil
+
+	return executor.ethereumClient.CheckRequiredBalance(ctx, token, amount)
 }
 
-func (executor *bridgeExecutor) isMintBurnToken(ctx context.Context, erc20Address common.Address) (bool, error) {
-	isMintBurnOnEthereum := executor.isMintBurnOnEthereum(ctx, erc20Address)
-	isMintBurnOnMultiversX := executor.isMintBurnOnMultiversX(ctx, erc20Address)
+func (executor *bridgeExecutor) isMintBurnToken(ctx context.Context, token common.Address, convertedToken []byte) (bool, error) {
+	isMintBurnOnEthereum := executor.isMintBurnOnEthereum(ctx, token)
+	isMintBurnOnMultiversX := executor.isMintBurnOnMultiversX(ctx, convertedToken)
 	if isMintBurnOnEthereum != isMintBurnOnMultiversX {
-		return false, ErrInvalidSetupMintBurnToken
+		return false, fmt.Errorf("%w isMintBurnOnEthereum = %v, isMintBurnOnMultiversX = %v", ErrInvalidSetupMintBurnToken, isMintBurnOnEthereum, isMintBurnOnMultiversX)
 	}
 	return isMintBurnOnEthereum, nil
 }
 
-func (executor *bridgeExecutor) checkRequiredMintBurnBalance(ctx context.Context, erc20Address common.Address) error {
-	mintedBalance, err := executor.ethereumClient.TokenMintedBalances(ctx, erc20Address)
+func (executor *bridgeExecutor) checkRequiredMintBurnBalance(ctx context.Context, token common.Address, convertedToken []byte) error {
+	mintedBalance, err := executor.ethereumClient.TokenMintedBalances(ctx, token)
 	if err != nil {
 		return err
 	}
 
-	burntBalance, err := executor.multiversXClient.AccumulatedBurnedTokens(ctx, erc20Address)
+	burntBalance, err := executor.multiversXClient.AccumulatedBurnedTokens(ctx, convertedToken)
 	if err != nil {
 		return err
 	}
-	// convert burntBalance to big.Int
-	burntBalanceBigInt := big.NewInt(0).SetUint64(burntBalance)
-	if mintedBalance.Cmp(burntBalanceBigInt) < 0 {
-		return fmt.Errorf("%w, minted: %s, burnt: %s for ERC20 token %s",
-			ErrMintBurnBalance, mintedBalance.String(), burntBalanceBigInt.String(), erc20Address.String())
+	if mintedBalance.Cmp(burntBalance) != 0 {
```

Comment:
is this ok? When will the burn be completed? At the moment of transfer in the mvx safe contract or when the transfer is completed on eth side and marked as finished on mvx?

Reply:
when the transfer is completed on eth side and marked as finished on mvx, thats why it should be equal at the moment of the batch execution

Reply:
ok, should work then